scala_proto/scala_proto.bzl

```diff
@@ -445,20 +445,12 @@ def scalapb_proto_library(
     )
 
     external_deps = list(SCALAPB_DEPS + GRPC_DEPS if (with_grpc) else SCALAPB_DEPS)
-
-    if with_java:
-        java_proto_lib = name + "_java_lib"
-        native.java_proto_library(
-            name = java_proto_lib,
-            deps = deps,
-            visibility = visibility,
-        )
-        external_deps.append(java_proto_lib)
+    scala_lib_deps = external_deps + deps
```

Comment:
do you not need the transitive compile time dependencies of any of the jvm code?

Also, I think we want to filter this so that we only include the non-proto deps. Otherwise, casual formatting changes to the proto that don't change the compile output will cause a recompilation.

Reply:
Two questions:

1. How can I get the transitive compile time deps for a jvm dep?
2. What is the best way to do this in a macro? Remember that this is not within a rule implementation so `deps` is just a list of strings. Should I use `native.existing_rule`? https://docs.bazel.build/versions/master/skylark/lib/native.html#existing_rule